MapboxVision/src/main/java/com/mapbox/vision/sensors/SensorsManager.kt

```diff
@@ -77,32 +79,35 @@ internal class SensorsManager(application: Application) : SensorEventListener {
             return
         }
         synchronized(this) {
-            lastTimestamp = event.timestamp / 1000000
+            lastTimestamp = TimeUnit.NANOSECONDS.toMillis(event.timestamp)
             when (event.sensor.type) {
-                TYPE_ACCELEROMETER -> {
-                    userAccelerationWithGravity[0] = event.values[0]
-                    userAccelerationWithGravity[1] = event.values[1]
-                    userAccelerationWithGravity[2] = event.values[2]
+                Sensor.TYPE_ACCELEROMETER -> {
+                    userAccelerationWithGravity.copyFrom(event.values)
                 }
-                TYPE_MAGNETIC_FIELD -> {
-                    geomagnetic[0] = event.values[0]
-                    geomagnetic[1] = event.values[1]
-                    geomagnetic[2] = event.values[2]
+                Sensor.TYPE_MAGNETIC_FIELD -> {
+                    geomagneticXyz.copyFrom(event.values)
+
+                    if (SensorManager.getRotationMatrix(
+                            rotationMatrix,
```

Comment:
So single rotationMatrix is used now?

Reply:
I'd suggest separate ones

Reply:
actually there's a few places where this matrix is used. It's not a problem, because it's like a _buffer_ for each orientation